src/test/HelmetTest.js

```diff
@@ -1256,6 +1321,52 @@ describe("Helmet", () => {
                 .that.equals(stringifiedScriptTags);
         });
 
+        it("will render html attributes as component", () => {
+            ReactDOM.render(
+                <Helmet
+                    htmlAttributes={{
+                        lang: "fr"
+                    }}
+                />,
+                container
+            );
+
+            const {htmlAttributes} = Helmet.rewind();
+            const attrs = htmlAttributes.toComponent();
+
+            expect(attrs).to.exist;
+
+            const markup = ReactServer.renderToStaticMarkup(
+                <html {...attrs}>
+                </html>
+            );
+
+            expect(markup)
+                .to.be.a("string")
+                .that.equals(`<html lang="fr"></html>`);
+        });
+
+        it("will render html attributes as string", () => {
+            ReactDOM.render(
+                <Helmet
+                    htmlAttributes={{
+                        lang: "ga",
+                        amp: undefined
+                    }}
+                />,
+                container
+            );
+
+            const head = Helmet.rewind();
+
+            expect(head.htmlAttributes).to.exist;
+            expect(head.htmlAttributes).to.respondTo("toString");
+
+            expect(head.htmlAttributes.toString())
+                .to.be.a("string")
+                .that.equals(`lang="ga" amp `);
```

Comment:
The other tests in `server` use the same string constant and show that pulling as a component or string gets you the same result.  Once you move the stringified html tag to a constant at the top, you can get this test to print into an html tag and make sure it also matches the same constant.